lib/src/metta/runner/modules/mod.rs

```diff
@@ -32,7 +65,6 @@ pub struct MettaMod {
     space: DynSpace,
     tokenizer: Shared<Tokenizer>,
     imported_deps: Mutex<HashMap<ModId, DynSpace>>,
```

Comment:
Are you talking about `imported_deps` storing `MettaMod` pointers rather than `DynSpace`?

The reason it works like this is because the space that's stored is a copy of the space of the child module, not just a pointer to the same underlying space.  The reason is described here: https://github.com/luketpeterson/hyperon-experimental/blob/4d0a92fd6032616a218713246b4f64520365f606/lib/src/metta/runner/modules/mod.rs#L262C1-L263C1  My current thinking regarding a clean fix is here: https://github.com/trueagi-io/hyperon-experimental/issues/618 in the "Different way to embed sub-module spaces"

Reply:
> Are you talking about imported_deps storing MettaMod pointers rather than DynSpace?

Not exactly. I was thinking about keeping just a `HashSet<ModId>`. But I looked at `imported_deps` usages and I see that this logic is used to insert stripped space as an atom into the space of the current module.